### DISCUSSION TITLE
Survival analysis with any regressor

### DISCUSSION THREAD

**Discussion by thierrymoudiki** [sebp/scikit-survival#493] [Ideas]
Hi,

I implemented this (work in progress): a way to use **any regressor (including in boosting) having `fit` and `predict`**.

https://github.com/thierrymoudiki/scikit-survival/blob/master/examples/2024-11-14-scikit-survival.ipynb

What's your opinion? Is this something that could be on your agenda, or should I follow my own route? 

T.

**Comment by sebp** [comment]
Hello and thanks for reaching out.

Can you explain a little bit what your thought process was to allow for other regressors? Is this based on a particular paper?

**Comment by thierrymoudiki** [comment]
Sure. Here's my thought process: 

1. First, I thought **it was possible** (because I use to do similar things for supervised learning and time series forecasting)
2. I searched for an 'entry point' for a long time, that I found in your Gradient Boosting Analysis, and replaced the base learner by _any_ model having `fit` and `predict`
3.  For the `Custom` class,  I relied on the Gradient Boosting implementation: only one boosting iteration is used, and a learning rate = 1.0

Not based on a paper at all (**yet**, but could be in the future)

**Comment by thierrymoudiki** [comment]
Follow-up: `survivalist`: Probabilistic model-agnostic survival analysis using scikit-learn, glmnet, xgboost, lightgbm, pytorch, keras, nnetsauce and mlsauce

https://thierrymoudiki.github.io/blog/2024/12/15/python/agnostic-survival-analysis
